src/ascent/python/CMakeLists.txt

```diff
@@ -42,62 +42,64 @@
 #
 ###############################################################################
 
-# Specify the sources of the pure python and compiled portions of our module.
-SET(ascent_py_python_sources  py_src/__init__.py py_src/ascent.py )
-#SET(ascent_py_headers        ascent_python.hpp)
-SET(ascent_py_cpp_sources     ascent_python.cpp)
-
-
-# Setup our module
-PYTHON_ADD_HYBRID_MODULE(ascent_python
-                         python-modules
-                         ascent
-                         setup.py
-                         "${ascent_py_python_sources}"
-                         #${ascent_py_headers}
-                         ${ascent_py_cpp_sources})
-
-# link with the proper libs (beyond python)
-target_link_libraries(ascent_python ascent)
-
-
-# TODO: ADD SUPPORT FOR SUPPORT PYTHON_MODULE_INSTALL_PREFIX
-# install the capi header so other python modules can use it
-# support alt install dir for python module via PYTHON_MODULE_INSTALL_PREFIX
-# if(PYTHON_MODULE_INSTALL_PREFIX)
-#     install(FILES ${ascent_py_headers} DESTINATION ${PYTHON_MODULE_INSTALL_PREFIX}/ascent/)
-# else()
-#     install(FILES ${ascent_py_headers} DESTINATION python-modules/ascent/)
-# endif()
+if (ENABLE_SERIAL)
```

Comment:
on develop we just changed how the python modules are created, there is one module that supports both the non-mpi and mpi case. 

You will have to change things a bit to get what you  are aiming for.

I think you can break out the non-mpi case to look like the mpi case - where you use a marco  to create only the compiled part of the python module.

https://github.com/Alpine-DAV/ascent/blob/4b5cf30a39690e2aa8d843edb38a7da376c984b1/src/ascent/python/ascent_module/CMakeLists.txt#L78

You will then need to modify current hybrid python module marco call to just be the macro call that uses distutils.

Additionally you can add error messages when the non-mpi compiled module fails to import in
https://github.com/Alpine-DAV/ascent/blob/develop/src/ascent/python/ascent_module/py_src/ascent.py

that mirror what is in:

https://github.com/Alpine-DAV/ascent/blob/develop/src/ascent/python/ascent_module/py_src/mpi/ascent_mpi.py

Reply:
Not sure I did this, lets see what the tests do.